processing/src/main/java/org/apache/druid/query/aggregation/constant/LongConstantAggregator.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.query.aggregation.constant;
+
+import org.apache.druid.query.aggregation.Aggregator;
+
+public class LongConstantAggregator implements Aggregator
```

Comment:
It might be worth making it so these constant aggregators use the no-op aggregators as base classes, so that only the get methods need to be overridden, to let these implementations be a bit more concise.

Reply:
Ack

Reply:
Those No-op aggregators are final classes so can't be extended. I don't see a clear reason why they are declared final. do you?

Reply:
If we rewrite the query to not include `GroupingAggregatorFactory` as I commented below, these aggregators won't be no longer in use.  `GroupingAggregatorFactory` can throw an exception instead when `factorize()` is called.